akka-cluster-sharding-typed/src/test/scala/akka/cluster/sharding/typed/JoinConfigCompatCheckerClusterShardingSpec.scala

```diff
@@ -32,6 +32,7 @@ object JoinConfig {
       akka.coordinated-shutdown.terminate-actor-system = on
       akka.remote.classic.netty.tcp.port = 0
       akka.remote.artery.canonical.port = 0
+      akka.actor.testkit.typed.system-shutdown-default = 10s # the default
```

Comment:
One might assume something in the stack is overriding this default to set a lower duration, from an extended test `Config` or test conf file or via one from any of these, however in hunting I did not find the culprit. Explicitly setting the default value fixes it locally, will see what Jenkins thinks.
```scala
lazy val clusterShardingTyped = akkaModule("akka-cluster-sharding-typed")
  .dependsOn(
    clusterTyped % "compile->compile;test->test;multi-jvm->multi-jvm",
    clusterSharding,
    actorTestkitTyped % "test->test",
    actorTypedTests % "test->test",
    persistenceTyped % "test->test",
    remoteTests % "test->test",
    jackson % "test->test")
```

Reply:
This is not the reason. 10s is already used according to log:
```
RuntimeException: Failed to stop [JoinConfigCompatibilitySpec] within [10 seconds]
```

I don't know what the reason could be but there is something with the leaving task in CoordinatedShutdown:
```
2020-01-12 12:40:44,458 WARN  akka.actor.CoordinatedShutdown  - Task [leave] failed in phase [cluster-leave]: Ask timed out on [Actor[akka://JoinConfigCompatibilitySpec/system/cluster#1040903017]] after [5000 ms]. Message of type [akka.cluster.CoordinatedShutdownLeave$LeaveReq$] was sent by [Actor[akka://JoinConfigCompatibilitySpec/system/cluster#1040903017]]. A typical reason for `AskTimeoutException` is that the recipient actor didn't send a reply. {akkaAddress=akka.tcp://JoinConfigCompatibilitySpec@127.0.1.1:43487, sourceThread=JoinConfigCompatibilitySpec-akka.actor.internal-dispatcher-27, akkaSource=CoordinatedShutdown(akka://JoinConfigCompatibilitySpec), sourceActorSystem=JoinConfigCompatibilitySpec, akkaTimestamp=11:40:44.457UTC}
2020-01-12 12:40:44,458 WARN  akka.actor.CoordinatedShutdown  - Coordinated shutdown phase [cluster-leave] timed out after 5000 milliseconds {akkaAddress=akka.tcp://JoinConfigCompatibilitySpec@127.0.1.1:43487, sourceThread=JoinConfigCompatibilitySpec-akka.actor.internal-dispatcher-23, akkaSource=CoordinatedShutdown(akka://JoinConfigCompatibilitySpec), sourceActorSystem=JoinConfigCompatibilitySpec, akkaTimestamp=11:40:44.458UTC}
```

I would also expect a log of
```
Marked address [{}] as [`Leaving`]
```
but that is missing.

Reply:
Thanks, I'll continue to look.

Reply:
It's just odd that the other very similar test with the same base test class runs without failure and there's very little difference. Also odd is that explicitly setting the test timing does make it pass now where without that it fails.

Reply:
> explicitly setting the test timing does make it pass now where without that it fails.

It doesn't fail consistently, though, right? I'm not opposed to this change but I'm not convinced it'd make any difference, either.

Reply:
so you mean that you can reproduce it (sometimes) locally? a log with debug level might be interesting to look at

Reply:
Once locally, before making any changes. Now never.

Reply:
Even squeezing some timeouts in config I can not get this to fail.
I am not sure the test itself is needed because the real test is for the incompatible config - to not allow a join. Which has no issues in typed or classic.